### PR TITLE
Adding histoSysEnvelope functionality 

### PR DESCRIPTION
**Pull request by fominik**
Hi
Following the discussion in issues here (sorry for the delay):
https://github.com/histfitter/histfitter/issues/215
I have tested that the tests still run fine, histograms are saved in the cache, and are added to the xml model. 

I didn't have an opportunity to compare the fit results of two different setups unfortunately.

Best
Nikolai

### CODE REVIEW

**Comment by longjon929** [comment]
Thanks @fominik , sorry for the indent trouble with the other files, I'm not sure why those weren't flagged before.  I'm taking a look at the tests

File: test/scripts/config_for_pytest.py [hunk 1]
```diff
@@ -138,9 +138,11 @@
 bkg1Sample.addSystematic( Systematic('weight_histoSysOneSide', configMgr.weights, ['weight', 'sys_weight4'], ['weight', 'sys_weight4'], 'weight', 'histoSysOneSide') )
 bkg1Sample.addSystematic( Systematic('weight_histoSysOneSideSym', configMgr.weights, ['weight', 'sys_weight5'], ['weight', 'sys_weight5'], 'weight', 'histoSysOneSideSym') )
 bkg1Sample.addSystematic( Systematic('weight_histoSysEnvelopeSym', configMgr.weights, ['weight', 'sys_weight6'], ['weight', '1/sys_weight6'], 'weight', 'histoSysEnvelopeSym') )
+bkg1Sample.addSystematic( Systematic("weight_histoSysEnvelope", configMgr.weights, ['weight','vector_of_weights',), ("6"), "weight", "histoSysEnvelope") )
```
Comment (longjon929) [inline test/scripts/config_for_pytest.py:141]: Here there is a ,) instead of a ]

**Comment by fominik** [comment]
Hi @longjon929 

I think the error is due to this histogram not being in the dictionary here:
https://github.com/histfitter/histfitter/blob/master/test/test_fullchain.py#L23-L137
I am trying to understand and fix it now.

Cheers
Nikolai

**Comment by longjon929** [comment]
> Hi @longjon929
> 
> I think the error is due to this histogram not being in the dictionary here: https://github.com/histfitter/histfitter/blob/master/test/test_fullchain.py#L23-L137 I am trying to understand and fix it now.
> 
> Cheers Nikolai

Ah yes, that could be, there is a snippet at the top (https://github.com/histfitter/histfitter/blob/master/test/test_fullchain.py#L10) to dump the values from the histCache for that dictionary.

**Comment by fominik** [comment]
Things should be fixed now, sorry about this @longjon929 
I had to adapt genTree.C a bit and to add the new integrals to test_fullchain.py, but now the tests should be passing.

A simple example of the systematic is:  
bkg1Sample.addSystematic( Systematic("weight_histoSysEnvelope", configMgr.weights, ['weight','vector_of_weights'], ("6"), "weight", "histoSysEnvelope") )

where ("6") tells the systematic to only consider the first 6 entries of the "vector_of_weights" vector. The systematic then builds the 6 histograms corresponding to these entries, and selects max/min for the envelope.

Cheers
Nikolai

File: test/scripts/config_for_pytest.py (hunk 1 above)
Comment (fominik) [reply test/scripts/config_for_pytest.py]: Fixed, thanks

**Comment by longjon929** [comment]
Thanks @fominik !  Looks good, once you've undrafted it I will merge.

**Comment by fominik** [comment]
Done! @longjon929

**Pull request merged**